_posts/2024-04-05-Configurable-Automation-for-OpenSearch-ML-Use-Cases.md

```diff
@@ -0,0 +1,133 @@
+---
+
+In OpenSearch, to use machine learning (ML) offerings, such as semantic, hybrid, and multimodal search, you often have to grapple with complex setup and preprocessing tasks. Additionally, you must write verbose queries, which can be a time-consuming and error-prone process.
+
+In this blog post, we introduce the OpenSearch Flow Framework plugin, [released in version 2.13](https://opensearch.org/blog/2.13-is-ready-for-download/) and designed to streamline this cumbersome process. By using this plugin, you can simplify complex setups with just one click. We've provided automated templates, enabling you to create connectors, register models, deploy them, and register agents and tools through a single API call. This eliminates the complexity of calling multiple APIs and orchestrating setups based on the responses.
+
+## Before the Flow Framework plugin
+
+Previously, setting up semantic search involved the steps outlined in the [semantic search documentation](https://opensearch.org/docs/latest/search-plugins/semantic-search/):
+
+1. Create a connector for a remote model, specifying pre- and post-processing functions.
+1. Register an embedding model using the connector ID obtained in the previous step.
+1. Configure an ingest pipeline to generate vector embeddings using the model ID of the registered model.
+1. Create a k-NN index and add the pipeline created in the previous step.
+
+This complex setup typically required you to be familiar with the OpenSearch ML Commons APIs. However, we are simplifying this experience through the Flow Framework plugin. Let's demonstrate how the plugin simplifies this process using the preceding semantic search example.
+
+## With the Flow Framework plugin
+
+In this example, you will configure the `semantic_search_with_cohere_embedding_query_enricher` workflow template. The workflow created using this template performs the following configuration steps:
+
+* Deploys an externally hosted Cohere model
+* Creates an ingest pipeline using the model
+* Creates a sample k-NN index and configures a search pipeline to define the default model ID for that index
+
+### Step 1: Create and provision the workflow
+
+Using the `semantic_search_with_cohere_embedding_query_enricher` workflow template, you provision the workflow with just one required field---the API key for the Cohere Embed model:
+
+```json
+POST /_plugins/_flow_framework/workflow?use_case=semantic_search_with_cohere_embedding_query_enricher&provision=true
+{
+    "create_connector.credential.key" : "<YOUR API KEY>"
+}
+```
+
+OpenSearch responds with a unique workflow ID, simplifying the tracking and management of the setup process:
+
+```json
+{
+  "workflow_id" : "8xL8bowB8y25Tqfenm50"
+}
+```
+
+Note: The workflow in the previous step creates a default k-NN index. The default index name is `my-nlp-index`:
+
+```json
+{
+  "create_index.name": "my-nlp-index"
```

Comment:
His is confusing having this separated from the line 47 API, particularly with the response JSON in between.  I *think* this goes in the block under line 49, but even I'm not sure.   Make it clear, perhaps including it as a line 50 after describing that it's optional, or perhaps repeating the whole REST call with both lines.

Reply:
I will remove the json completely